packages/components/src/custom-select/index.tsx

```diff
@@ -0,0 +1,37 @@
+/**
+ * External dependencies
+ */
+// eslint-disable-next-line no-restricted-imports
+import * as Ariakit from '@ariakit/react';
+
+/**
+ * Internal dependencies
+ */
+import {
+	CustomSelectButton,
+	CustomSelectPopover,
+	CustomSelectItem as SelectItem,
+} from './styles';
+import { type CustomSelectProps } from './types';
+
+export function CustomSelect( props: CustomSelectProps ) {
+	const { label, onChange, options } = props;
+
+	const store = Ariakit.useSelectStore( {
+		setValue: ( value ) => onChange?.( value ),
```

Comment:
Our current `CustomSelectControl` expects an object to be returned via the `onChange` event: 

https://github.com/WordPress/gutenberg/blob/887cf3dfc4b3426ce5696a383216186233c468e1/packages/components/src/custom-select-control/index.js#L100-L107

Which is managed by `useSelect`; part of the package `downshift`. 

In Ariakit, [`setValue` appears to be what we'd use in lieu of `useSelect`](https://ariakit.org/reference/use-select-store#setvalue),  but `value`/`setValue` expect a string (or an array of strings for multi-select).

Reply:
It appears that the object passed to `onSelectItemChange` has the following structure:

```js
{
  selectedItem: "Curium",
  type: "__item_click__",
  highlightedIndex: -1,
  isOpen: false,
  inputValue: "",
}
```

I'm not sure if all the elements in this structure are part of the public API. The `type` value, in particular, seems obscure. I'm also unsure about the meaning of `highlightedIndex`, as it always appears to be `-1`.

Regardless, everything except for `type` can be readily accessed from within the `setValue` callback. You can use `store.getState()` in this context as well:

```jsx
const store = useSelectStore({
  async setValue(selectedItem) {
    if (!onChange) return;
    // Executes the logic in a microtask after the popup is closed. This might not be necessary.
    // This is simply to ensure the isOpen state matches that in Downshift.
    await Promise.resolve();
    const state = store.getState();
    const changeObject = {
      selectedItem,
      type: "__item_click__",
      highlightedIndex: state.renderedItems.findIndex((item) => item.value === selectedItem),
      isOpen: state.open,
      inputValue: "",
    };
    onChange(changeObject);
  },
});
```

Reply:
Thank you for this! We finally have gotten a chance to try this out. 🙂